src/controller.rs

```diff
@@ -421,6 +425,14 @@ impl Controller {
                 for x in offsets {
                     ::log::info!("Setting offset at {:.4}: {:.4} (cost {:.4})", x.0, x.1, x.2);
                     let new_ts = ((x.0 - x.1) * 1000.0) as i64;
+                    //Here the offsets are checked
+                    if !this.sync_rank.is_empty(){
+                        let index = ((x.0 - x.1) as f64 / (this.sync_ratio*1000.0)).round() as usize;
+                        if index < this.sync_rank.len() && this.sync_rank[index] < 20.0{
+                            println!("be={}",this.sync_rank[index]);

```

Comment:
if this log is useful, please use `log::debug!` instead of `println`
if it's not useful for most use cases then just remove it